validator/keymanager/local/import.go

```diff
@@ -49,52 +62,67 @@ func (km *Keymanager) ImportKeystores(
 			log.Error(err)
 		}
 		// if key exists prior to being added then output log that duplicate key was found
-		if _, ok := keys[string(pubKeyBytes)]; ok {
+		_, isDuplicateInArray := keys[string(pubKeyBytes)]
+		_, isDuplicateInExisting := existingPubKeys[string(pubKeyBytes)]
+		if isDuplicateInArray || isDuplicateInExisting {
 			log.Warnf("Duplicate key in import will be ignored: %#x", pubKeyBytes)
 			statuses[i] = &ethpbservice.ImportedKeystoreStatus{
 				Status: ethpbservice.ImportedKeystoreStatus_DUPLICATE,
 			}
 			continue
 		}
+
 		keys[string(pubKeyBytes)] = string(privKeyBytes)
 		statuses[i] = &ethpbservice.ImportedKeystoreStatus{
 			Status: ethpbservice.ImportedKeystoreStatus_IMPORTED,
 		}
+		totalImported++
 	}
-	privKeys := make([][]byte, 0)
-	pubKeys := make([][]byte, 0)
-	for pubKey, privKey := range keys {
-		pubKeys = append(pubKeys, []byte(pubKey))
-		privKeys = append(privKeys, []byte(privKey))
+	if totalImported == 0 {
+		log.Warn("no keys were imported")
+		return statuses, nil
 	}
-
-	// Write the accounts to disk into a single keystore.
-	accountsKeystore, err := km.CreateAccountsKeystore(ctx, privKeys, pubKeys)
-	if err != nil {
-		return nil, err
+	// 2) Import the keys into copied in memory keystore,clear duplicates in existing set
+	// duplicates,errored ones are already skipped
+	for pubKey, privKey := range keys {
+		storeCopy.PublicKeys = append(storeCopy.PublicKeys, []byte(pubKey))
+		storeCopy.PrivateKeys = append(storeCopy.PrivateKeys, []byte(privKey))
 	}
-	encodedAccounts, err := json.MarshalIndent(accountsKeystore, "", "\t")
-	if err != nil {
+	//3 & 4) save to disk and re-initializes keystore
+	if err := km.SaveStoreAndReInitialize(ctx, storeCopy); err != nil {
 		return nil, err
 	}
-	if err := km.wallet.WriteFileAtPath(ctx, AccountsPath, AccountsKeystoreFileName, encodedAccounts); err != nil {
-		return nil, err
+	// 5) Verify keys are indeed Imported
+	if len(km.accountsStore.PublicKeys) < len(storeCopy.PublicKeys) {
+		return nil, fmt.Errorf("keys were not imported successfully, expected %d got %d", len(storeCopy.PublicKeys), len(km.accountsStore.PublicKeys))
 	}
+	//
+	// 6) Return Statuses
 	return statuses, nil
 }
 
 // ImportKeypairs directly into the keymanager.
```

Comment:
I think this could be updated. This routine does more than import directly into km

Reply:
do you suggest breaking this function up?

Reply:
No, I think it's already split up well. I think the comment could mention how this first attempts to save to disk then updates the keymanager rather than "imports directly into keymanager"